src/main/java/org/alfresco/repo/search/impl/querymodel/impl/db/DBQueryEngine.java

```diff
@@ -205,19 +206,39 @@ public QueryEngineResults executeQuery(Query query, QueryOptions options, Functi
         dbQuery.setSinceTxId(sinceTxId);
         
         dbQuery.prepare(namespaceService, dictionaryService, qnameDAO, nodeDAO, tenantService, selectorGroup, null, functionContext, metadataIndexCheck2.getPatchApplied());
+        dbQuery.setMaxItems(options.getMaxItems() + 1);
+        dbQuery.setSkipCount(options.getSkipCount());
         List<Node> nodes = template.selectList(SELECT_BY_DYNAMIC_QUERY, dbQuery);
-        LinkedHashSet<Long> set = new LinkedHashSet<Long>(nodes.size());
+        LinkedHashSet<Long> set = new LinkedHashSet<Long>(nodes.size());
         for(Node node : nodes)
         {
             set.add(node.getId());
         }
         List<Long> nodeIds = new ArrayList<Long>(set);
-        ResultSet rs =  new DBResultSet(options.getAsSearchParmeters(), nodeIds, nodeDAO, nodeService, tenantService, Integer.MAX_VALUE);
-        ResultSet paged = new PagingLuceneResultSet(rs, options.getAsSearchParmeters(), nodeService);
+        
+        SearchParameters params = options.getAsSearchParmeters();
+        params.setSkipCount(0);
+        
+        DBResultSet rs =  new DBResultSet(params, nodeIds, nodeDAO, nodeService, tenantService, Integer.MAX_VALUE);
+        rs.setNumberFound(computeNumberFound(options, nodeIds.size()));
+        
+        PagingLuceneResultSet paged = new PagingLuceneResultSet(rs, params, nodeService);
         
         answer.put(key, paged);
         return new QueryEngineResults(answer);
     }
+
+    private int computeNumberFound(QueryOptions options, int numberOfIds)
+    {
+        int res;
+
+        if (numberOfIds > 0)
+            res = options.getSkipCount() + numberOfIds;
+        else
+            res = 0;
```

Comment:
ACS code formatting requires braces everywhere even if not explicitly required.
See https://hub.alfresco.com/t5/alfresco-content-services-hub/coding-standards-for-alfresco-content-services/ba-p/290457#toc-hId-1652896845